frontend/src/static/js/v_summary.js

```diff
@@ -465,13 +465,20 @@ window.vSummary = {
     mergeGroup() {
       this.filtered.forEach((group, groupIndex) => {
         const dateToIndexMap = {};
+        const dailyIndexMap = {};
         const mergedCommits = [];
+        const mergedDailyCommits = [];
         const mergedFileTypeContribution = {};
         let mergedVariance = 0;
         let totalMergedCommits = 0;
 
         group.forEach((user) => {
-          this.mergeCommits(user, mergedCommits, dateToIndexMap);
+          user.commits.forEach((commit) => {
+            this.mergeCommits(commit, user, dateToIndexMap, mergedCommits);
+          });
+          user.dailyCommits.forEach((commit) => {
```

Comment:
From my discussion with you last time, this extra step is so that you can get the `trueCommittedDate` for commits with granularity `week`? How about we just have an extra property `trueCommittedDate` for those commits?

Reply:
The problem is the ramp.pug file and the difference in structure of commits. The ramp.pug will try to read commit.date. When set to week, the commits belong to the same week will all be in commit.commitResult. We thus cannot read the true date of each commit unless we change all related code in ramp and zoom